.github/workflows/build-images-docs-builder.yaml

```diff
@@ -0,0 +1,147 @@
+on:
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - Documentation/Dockerfile
+      - Documentation/requirements.txt
+
+permissions:
+  # To be able to access the repository with `actions/checkout`
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  build-and-push:
+    timeout-minutes: 45
+    environment: docs-builder
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout default branch (trusted)
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          persist-credentials: false
+
+      - name: Set environment variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4c0219f9ac95b02789c1075625400b2acbff50b1 # v2.9.1
+
+      # Warning: since this is a privileged workflow, subsequent workflow job
+      # steps must take care not to execute untrusted code.
+      - name: Checkout pull request branch (NOT TRUSTED)
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          persist-credentials: false
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up git
+        run: |
+          git config user.name "Cilium Imagebot"
+          git config user.email "noreply@cilium.io"
+
+      - name: Generate image tag for docs-builder
+        id: docs-builder-tag
+        run: |
+          echo tag="$(git ls-tree --full-tree HEAD -- ./Documentation | awk '{ print $3 }')" >> $GITHUB_OUTPUT
+
+      - name: Check if tag for docs-builder already exists
+        id: docs-builder-tag-in-repositories
+        shell: bash
+        run: |
+          if docker buildx imagetools inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ steps.docs-builder-tag.outputs.tag }} &>/dev/null; then
+            echo exists="true" >> $GITHUB_OUTPUT
+          else
+            echo exists="false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Login to quay.io
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_DOCS_BUILDER_USERNAME }}
+          password: ${{ secrets.QUAY_DOCS_BUILDER_PASSWORD }}
+
+      - name: Build docs-builder image
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        id: docker_build_docs_builder
+        with:
+          provenance: false
+          context: ./Documentation
+          file: ./Documentation/Dockerfile
```

Comment:
Is there a way to edit this Dockerfile to expose environment variables from the host and echo them to the GHA output?

Reply:
I guess with `environment: docs-builder` this requires docs-structure review anyway so as long as docs reviewers are aware that they need to check for env var abuse in any changes that are proposed against these files, then that is fine. Otherwise if docs-structure reviewers can't be trusted to enforce this then we'd need to think about how to ensure that folks with the appropriate knowledge are put in the path for running these checks.

Reply:
I also wasn't 100% sure whether the secrets are available if a PR is opened from an external fork. If it requires push access to cilium/cilium then that is also a mitigating factor, as that requires org-member privileges IIRC.

Reply:
> Is there a way to edit this Dockerfile to expose environment variables from the host and echo them to the GHA output?

Excellent question. As far as I'm aware, the only way would be to use the `ARG` directive in the Dockerfile, _**and**_ the `--build-arg` argument for the `docker` invocation; but the workflow file does not use the relevant `build-args` parameter for the `docker/build-push-action` action, so it shouldn't be possible to do it this way.

This being said, I'm not well versed on Docker security issues, so there could be something else I'm not aware of.

Reply:
This is correct. FYI we have the exact same setup in the image build workflows. These workflows run on `pull_request_target` so they run in a privileged context, and are indeed able to access repository secrets (and in general they have various dangerous permissions), so we must be careful as to what comes from the PR and how we use it.

In the case of `docker/build-push-action` using a Dockerfile coming from the PR it is fine because as said by Quentin, for the Dockerfile to be able to access the environment variables from the privileged workflow we would need to pass the secrets explicitly to the Dockerfile (as is done here: https://docs.docker.com/build/ci/github-actions/secrets/).